modules/reference/pages/data-transform-rust-sdk.adoc

```diff
@@ -0,0 +1,134 @@
+= Rust SDK for Data Transforms
+:description: Work with data transforms using Rust.
+
+Rust SDK reference documentation for Redpanda data transforms.
+
+The `redpanda_transform_sdk` crate contains the core functions and types for transforming data within Redpanda.
+
+== Functions
+
+The Rust SDK includes the following https://doc.rust-lang.org/rust-by-example/fn.html[function] to manage data transforms to an output topic.  
+
+=== on_record_written
+
+[,rust]
+----
+pub fn on_record_written<E, F>(cb: F) -> !
+where
+    E: Debug,
+    F: Fn(WriteEvent<'_>, &mut dyn RecordWriter) -> Result<(), E>,
+----
+
+The `on_record_written()` function transforms individual records after they have been written to an input topic. The function registers a callback to be fired when a record is written to the topic. This callback is triggered after the record has been written and fsynced to disk and the producer has been acknowledged (see <<writeevent,`WriteEvent`>>).
+
+Since this method blocks and runs forever, it should be called from the `main()` function. Perform the necessary steps before transforming any records before calling this method.
+
+---
+
+== Structs
+
+A https://doc.rust-lang.org/rust-by-example/custom_types/structs.html[struct] is a custom data type that allows you to define a structure composed of related data attributes.
+
+=== BorrowedHeader
+
+[,rust]
+----
+pub struct BorrowedHeader<'a> { /* private fields */ }
+----
+
+`BorrowedHeader` is a zero-copy <<borrowedrecord,`BorrowedRecord`>> header.
+
+=== BorrowedRecord
+
+[,rust]
+----
+pub struct BorrowedRecord<'a> { /* private fields */ }
+----
+
+`BorrowedRecord` is a zero-copy representation of a <<record,`Record`>> in Redpanda. 
+
+=== Record
+
+[,rust]
+----
+pub struct Record { /* private fields */ }
+----
+
+`Record` is a record in Redpanda. It consists of a key-value pair of bytes, along with a collection of <<recordheader,`RecordHeader`>>.
+
+Records are generated as the result of any transforms that act upon a `BorrowedRecord`.
+
+=== RecordHeader
+
+[,rust]
+----
+pub struct RecordHeader { /* private fields */ }
+----
+
+A `RecordHeader` is a key-value pair attached to a <<record,`Record`>>. Headers are opaque to the broker and are purely a mechanism for the producer and consumers to pass information.
+
+=== RecordWriter
+
+[,rust]
+----
+pub struct RecordWriter<'a> { /* private fields */ }
```

Comment:
how come we don't document the properties and methods for these structs. How should users know that `writer.record` or `writer.record.key()` exists?

Reply:
May have just missed this info, @rockwotj would you mind confirming if this has all been provided in https://docs.rs/redpanda-transform-sdk/0.2.0/redpanda_transform_sdk/index.html?

Reply:
Oh yes silly me. For each of these times we still need the methods for each struct/trait:

For RecordWriter:

https://docs.rs/redpanda-transform-sdk/0.2.0/redpanda_transform_sdk/struct.RecordWriter.html

You'll have to drill down to get more information.